projects/components/src/draggable-list/draggable-list.component.ts

```diff
@@ -39,14 +42,20 @@ export class DraggableListComponent<T> implements AfterContentInit {
   @ContentChildren(DraggableItemComponent)
   public draggableItemsRef!: QueryList<DraggableItemComponent<T>>;
 
-  public draggableItems: DraggableItemComponent<T>[] = [];
+  public draggableItems$!: Observable<DraggableItemComponent<T>[]>;
 
   public ngAfterContentInit(): void {
-    this.draggableItems = this.draggableItemsRef.toArray();
+    this.draggableItems$ = queryListAndChanges$(this.draggableItemsRef).pipe(
+      map(draggableItems => draggableItems.toArray())
+    );
   }
 
   public dropList(event: CdkDragDrop<DraggableItemComponent<unknown>[]>): void {
-    moveItemInArray(this.draggableItems, event.previousIndex, event.currentIndex);
-    this.draggableListChange.emit(this.draggableItems.map(dragabbleItem => dragabbleItem.data!));
+    this.draggableItems$ = this.draggableItems$.pipe(
```

Comment:
I find this a little confusing. If you just want to perform an action on the current draggable items then you could just use it directly from `draggableItemsRef`. queryListAndChanges will just run the CD automatically on any addition/deletion of draggable items.

Reply:
yeah - what we're doing here isn't right, draggableItems$ should never be reassigned. This would create a sequence of new observables (since each dropList invocation builds on the draggableItems$ produced by the previous), building more and more tap functions along the chain each of which is emitting different things based on the enclosed event.

Reply:
What if I subscribe to the draggableItems$ and modify it internally.

Reply:
I think I would start with Anand's suggestion - does the query list allow updates? If not, we probably want to use a subject and merge the query list and our own updates to create draggableItems

Reply:
@palbizu Would this work? 
```
public dropList(event: CdkDragDrop<DraggableItemComponent<unknown>[]>): void {
   const draggableItems = this.draggableItemsRef.toArray();
   moveItemInArray(draggableItems, event.previousIndex, event.currentIndex);
   this.draggableListChange.emit(draggableItems.map(dragabbleItem => dragabbleItem.data!))
}
```

Reply:
done